source/Nuke.Common/Tools/OctoVersion/OctoVersionTasks.cs

```diff
@@ -31,16 +40,18 @@ internal static string GetToolPath(string framework = null)
 
         private static OctoVersionInfo GetResult(IProcess process, OctoVersionGetVersionSettings toolSettings)
         {
+            ControlFlow.Assert(!string.IsNullOrEmpty(toolSettings.OutputJsonFile), $"{nameof(toolSettings.OutputJsonFile)} must be set");
+
+            var output = "";
             try
             {
-                var output = process.Output.EnsureOnlyStd().Select(x => x.Text).ToList();
+                output = File.ReadAllText(toolSettings.OutputJsonFile);
```

Comment:
`SerializationTasks.JsonDeserializeFromFile<JObject>` ... afaik, it does no exception handling, but it's on my list. Anyway, I don't expect any issues with the output file, so better to use the central API

Reply:
:+1:

Reply:
I looked into doing this but the central API can't handle the deserialization as we need to use a `AllWritableContractResolver`.  Our approach here is the same as what Nerdbank and GitVersion are already doing.

I propose we leave this as is, and if any future work occurs to enhance the central API to handle this case, all 3 version tools can be updated to use the central API?

Reply:
True, but maybe we can adapt that and add the resolver per default there too.

Alternatively – although not so clean – it could also be this:
```
SerializationTasks.JsonDeserialize(input, settings =>
{
    settings.ContractResolver = new AllWritableContractResolver();
    return settings;
});
```
```

Reply:
ive pushed an update that enhances the Nuke `SerializationTasks.JsonDeserializeFromFile()` to take a `Configurator` and used that